### PR TITLE
Flowmachine subscriber degree

### LINKED ISSUE #969: Expose SubscriberDegree via api
- [ ] Add marshmallow spec for `SubscriberDegree `
- [ ] Add to allowable joined aggregate queries
- [ ] Add flowclient function `subscriber_degree(*, start, stop, direction, subscriber_subset)`
- [ ] Add test to `integration_tests/tests/test_queries.py`
- [ ] Update acceptance tests
  - [ ] integration_tests/tests/flowapi_tests/test_api_spec.py
  - [ ] integration_tests/tests/flowmachine_server_tests/test_server.py
- [ ] Update `docs/source/4-developer.md` with newly added query
- [ ] Change returned column name to `value` from `degree` in `SubscriberDegree ` in SQL and `column_names` attribute
- [ ] Update tests using `SubscriberDegree ` to reflect changed column name
- [ ] Update Changelog

### PR DESCRIPTION
**Pull request by BhavinPanch**
Closes #969 

### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [x] Added tests for any new additions
- [ ] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [x] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

Exposed subscriber degree information via joined spatial aggregate of FlowAPI

### CODE REVIEW

File: flowmachine/flowmachine/core/server/query_schemas/subscriber_degree.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    query_kind = fields.String(validate=OneOf(["subscriber_degree"]))
+    start = fields.Date(required=True)
+    stop = fields.Date(required=True)
+    direction = fields.String(
+        required=False, validate=OneOf(["in", "out", "both"]), default="both"
+    )  # TODO: use a globally defined enum for this
+    subscriber_subset = SubscriberSubset()
+
+    @post_load
+    def make_query_object(self, params):
```
Comment (maxalbert) [inline flowmachine/flowmachine/core/server/query_schemas/subscriber_degree.py]: As of #990, this needs a `**kwargs` argument due to the dependency upgrade of marshmallow to `3.0.0.rc7` which [introduced](https://marshmallow.readthedocs.io/en/3.0/upgrading.html#decorated-methods-receive-many-and-partial) a backwards-incompatible change in methods decorated with `@post_load`.

Comment (maxalbert) [reply flowmachine/flowmachine/core/server/query_schemas/subscriber_degree.py]: (This is what's causing the currently failing integration test on CircleCI.)

Comment (BhavinPanch) [reply flowmachine/flowmachine/core/server/query_schemas/subscriber_degree.py]: That makes sense. I will change it now.

File: flowclient/flowclient/client.py [hunk 2]
```diff
@@ -1285,3 +1285,39 @@ def unique_location_counts(
...
+    """
+    Return query spec for subscriber degree
+
+    Parameters
+    ----------
+    start : str
+        ISO format date of the first day of the count, e.g. "2016-01-01"
+    stop : str
+        ISO format date of the day _after_ the final date of the count, e.g. "2016-01-08"
+    direction : {"in", "out", "both"}, default "both"
```
Comment (greenape) [inline flowclient/flowclient/client.py]: There isn't a default defined in the function here?

**Pull request merged**